File: Excel/594 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/594 Capitalize at Same Indexes.xlsx"
input = read_excel(path, range = "A1:A10")
test  = read_excel(path, range = "B1:B10")

result = input %>%
  mutate(pos = str_locate_all(Sentences, "[A-Z]")) %>%
  mutate(answer = str_to_lower(Sentences) %>%
           str_remove_all("[^a-z]") %>%
           str_split("")) %>%
  mutate(answer = map2(answer, pos, ~{
    .x[.y] = toupper(.x[.y]) 
    .x
  })) %>%
  mutate(answer = map_chr(answer, ~{
    paste(.x[!is.na(.x)], collapse = "")
  }))

all.equal(result$answer, test$`Answer Expected`)
#> [1] TRUE          

File: Excel/500-599/594/594 Challenge.py
import pandas as pd
import re

path = "594 Capitalize at Same Indexes.xlsx"
input = pd.read_excel(path, usecols="A", nrows=10)
test = pd.read_excel(path, usecols="B", nrows=10)

def capitalize_at_indexes(sentence):
    pos = [m.start() for m in re.finditer(r'[A-Z]', sentence)]
    answer = list(re.sub(r'[^a-zA-Z]', '', sentence).lower())
    answer = [char.upper() if i in pos else char for i, char in enumerate(answer)]
    return ''.join(answer)

input['answer'] = input['Sentences'].apply(capitalize_at_indexes)

print(input['answer'].equals(test['Answer Expected'])) # True
